libs/pika/async_mpi/include/pika/async_mpi/mpi_polling.hpp

```diff
@@ -26,17 +26,29 @@
 #include <vector>
 
 namespace pika::mpi::experimental {
+
+    enum stream_index : std::uint32_t
+    {
+        default_stream = 0,
+        send_stream,
+        receive_stream,
+        collective_stream,
+        user_stream,
+    };
```

Comment:
Maybe something like this?
```suggestion
    enum class stream_type : std::uint32_t // does the user need to know that it's used as an index? it's the type of stream that they're interested in, right?
    {
        default_ = 0, // possibly different name here to avoid the underscore...
        send,
        receive,
        collective,
        user,
    };
```

Reply:
I'm happy with that change - I used "index" in the name because if the user decided they needed more streams, then
`user_stream + 1`, `user_stream + 2`, etc might be needed. But I will worry about that some other time and see iif this is a viable way of doing things first.